src/NuGetPackageVerifier/Rules/PackageVersionMatchesAssemblyVersionRule.cs

```diff
@@ -6,39 +6,54 @@
 using System.Linq;
 using System.Reflection;
 using Mono.Cecil;
+using NuGet.Versioning;
 
 namespace NuGetPackageVerifier.Rules
 {
     public class PackageVersionMatchesAssemblyVersionRule : AssemblyHasAttributeRuleBase
     {
-        private Version _packageVersion;
+        private bool _isPrerelease;
+        private NuGetVersion _packageVersion;
         private string _packageId;
 
         public override IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
         {
-            _packageVersion = context.Metadata.Version.Version;
+            _isPrerelease = context.Metadata.Version.IsPrerelease;
```

Comment:
Hmm first time I'm noticing this, but I don't think this code is technically safe (though in practice it will happen to work). The problem is that the same package rule instances are used for all packages, so package rules need to avoid having per-package state. We should fix this in a separate PR (after this PR). I filed https://github.com/aspnet/BuildTools/issues/166 to track fixing this.